From abc7de46102bfc2766a5c22ce59807db3d5638bb Mon Sep 17 00:00:00 2001
Message-Id: <abc7de46102bfc2766a5c22ce59807db3d5638bb.1509410670.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 31 Oct 2017 01:35:34 +0100
Subject: [PATCH] Don't free instance_recur_id when not initialized

Caused chaos and confusing crashes in gnome-shell-calendar-server.

Fixes efe14e5959. Found via bisect and again by valgrind.
---
 src/calendar/libecal/e-cal-client.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/calendar/libecal/e-cal-client.c b/src/calendar/libecal/e-cal-client.c
index a4143ecd6749a884..546426243d35d008 100644
--- a/src/calendar/libecal/e-cal-client.c
+++ b/src/calendar/libecal/e-cal-client.c
@@ -2558,8 +2558,8 @@ process_detached_instances (GSList *instances,
 						ci->comp = comp;
 					}
 				}
+				e_cal_component_free_range (&instance_recur_id);
 			}
-			e_cal_component_free_range (&instance_recur_id);
 		}
 
 		e_cal_component_free_datetime (&recur_id.datetime);
-- 
2.14.2

